src/app/zap-templates/endpoint_config.zapt

```diff
@@ -3,4 +3,100 @@
 // Prevent multiple inclusion
 #pragma once
 
-// TODO issue #3637
+{{#endpoint_config}}
+
+
+// Default values for the attributes longer than a pointer,
+// in a form of a binary blob
+// Separate block is generated for big-endian and little-endian cases.
+#if BIGENDIAN_CPU
+#define GENERATED_DEFAULTS {{endpoint_attribute_long_defaults endian="big"}}
+
+#else // !BIGENDIAN_CPU
+#define GENERATED_DEFAULTS {{endpoint_attribute_long_defaults endian="little"}}
+#endif // BIGENDIAN_CPU
+
+#define ZAP_TYPE(type) ZCL_ ## type ## _ATTRIBUTE_TYPE
+#define ZAP_LONG_DEFAULTS_INDEX(index) {(uint8_t*)(&generatedDefaults[index])}
+#define ZAP_MIN_MAX_DEFAULTS_INDEX(index) {(uint8_t*)(&minMaxDefault[index])}
+
+// This is an array of EmberAfAttributeMinMaxValue structures.
+#define GENERATED_MIN_MAX_DEFAULT_COUNT {{endpoint_attribute_min_max_count}}
+#define GENERATED_MIN_MAX_DEFAULTS {{endpoint_attribute_min_max_list}}
+
+#define ZAP_ATTRIBUTE_MASK(mask) ATTRIBUTE_MASK_ ## mask
+// This is an array of EmberAfAttributeMetadata structures.
+#define GENERATED_ATTRIBUTE_COUNT {{endpoint_attribute_count}}
+#define GENERATED_ATTRIBUTES {{ endpoint_attribute_list }}
+
+// This is an array of EmberAfCluster structures.
+#define ZAP_ATTRIBUTE_INDEX(index) ((EmberAfAttributeMetadata *)(&generatedAttributes[index]))
+
+#define GENERATED_CLUSTER_COUNT {{endpoint_cluster_count}}
+#define GENERATED_CLUSTERS {{endpoint_cluster_list}}
```

Comment:
@vivien-apple in that case a new issue should be open to fix this

Reply:
It does not sounds right to just remove methods clusters relies on to works properly.
What is the issue to generate those exactly ? I assume that's the same root issue than plugins ?

Reply:
Long story short these methods and array are silabs specific and have no direct link with the ZCL part. Therefore, ZAP cannot generate them. However, we have a SDK extensions feature in ZAP that we could use to generate those arrays and link them. So either we modify the zcl framework to fit this new reality or we use the sdk extension to generate the `GENERATED_FUNCTIONS_ARRAYS` either way it should be done in a separate issue because of the amount of work required.

Reply:
> Long story short these methods and array are silabs specific and have no direct link with the ZCL part. Therefore, ZAP cannot generate them. However, we have a SDK extensions feature in ZAP that we could use to generate those arrays and link them. So either we modify the zcl framework to fit this new reality or we use the sdk extension to generate the `GENERATED_FUNCTIONS_ARRAYS` either way it should be done in a separate issue because of the amount of work required.

The issue is that https://github.com/project-chip/connectedhomeip/blob/515d2ccc76736eff344e5fd2acd0ef25a7863cbc/examples/lighting-app/nrfconnect/main/ZclCallbacks.cpp#L87 depends on such mechanism.

Maybe that's something we need to fix first. Do you mind pointing me to the "SDK extensions feature" that you mentioned ?

Reply:
Here's the documentation about the SDK. https://github.com/project-chip/zap/blob/master/docs/sdk-integration.md

Reply:
> Here's the documentation about the SDK. https://github.com/project-chip/zap/blob/master/docs/sdk-integration.md

Do you have a direct link to an example of the SDK feature that may help us to address this use case ? It would make my life easier instead of having to found something in the whole SDK documentation :)

Reply:
I do not have a good answer yet in order to solve the mentioned issue. What do you think of hardcoding this part in the template at the moment and fill a followup ? It will let us move forward.

Reply:
A partly Hard-coded solution is now implemented to generate the `GENERATED_FUNCTIONS` macro